toolvoice/scripts/inference/openai_realtime_infer.py

```
@@ -0,0 +1,386 @@
+
+async def run_row_when2call(ws_url, headers, chunk_reader, audio_path, tool_cell) -> Dict[str, Any]:
+    assistant_text, tool_calls = await run_realtime_session(ws_url, headers, chunk_reader, audio_path, tool_cell, WHEN2CALL_INSTRUCTIONS)
+
+    if tool_calls:
+        return {"response_type": "tool_call", "response": json.dumps(tool_calls, ensure_ascii=False)}
+
+    cleaned = assistant_text.replace("```", "").replace("json", "")
+    try:
+        parsed = json.loads(cleaned)
+        return {"response_type": parsed.get("Type"), "response": parsed.get("Response")}
+    except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)